### PR TITLE
Unwrap incoming kafka message following OrderEvent avro schema

### PR DESCRIPTION
**Pull request by vuldin**
**Work in progress This PR will be finalized once tests are complete**

- using `order` topic
- made multiple props prod-only

### Start consumer
1. set `KAFKA_TOPIC=orders` environment variable in a terminal
2. change into the directory `quarkus-reactive-quarkus-consumer`
3. `quarkus dev`

### Start producer
1. change into the directory `quarkus-reactive-quarkus-producer`
2. `quarkus dev`

Go to the following page to see the channel (and topic) details: http://localhost:8080/q/dev/io.quarkus.quarkus-smallrye-reactive-messaging/channels

### CODE REVIEW

File: quarkus-reactive-kafka-consumer/pom.xml [hunk 1]
```diff
@@ -16,6 +16,9 @@
     <quarkus.platform.group-id>io.quarkus.platform</quarkus.platform.group-id>
     <quarkus.platform.version>2.5.2.Final</quarkus.platform.version>
     <surefire-plugin.version>3.0.0-M5</surefire-plugin.version>
+    <!--
+    <apicurio.version>2.1.4.Final</apicurio.version>
+    -->
```
Comment (vuldin) [inline quarkus-reactive-kafka-consumer/pom.xml:21]: Need to delete these lines.

File: quarkus-reactive-kafka-consumer/pom.xml [hunk 2]
```diff
@@ -51,17 +54,32 @@
...
+    </dependency>
+    -->
   </dependencies>
   <build>
     <plugins>
       <plugin>
         <groupId>io.quarkus</groupId>
         <artifactId>quarkus-maven-plugin</artifactId>
-        <version>${quarkus-plugin.version}</version>
+        <version>${quarkus.platform.version}</version>
```
Comment (vuldin) [inline quarkus-reactive-kafka-consumer/pom.xml:82]: While this resolved the issue, I'm not sure why this change was needed in the first place. My assumption is `quarkus-plugin.version` is from a previous quarkus version.

File: quarkus-reactive-kafka-consumer/pom.xml [hunk 3]
```diff
@@ -51,17 +54,32 @@
...
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-apicurio-registry-avro</artifactId>
+    </dependency>
+    <!--
+    <dependency>
+      <groupId>io.apicurio</groupId>
+      <artifactId>apicurio-registry-utils-serde</artifactId>
+      <version>${apicurio.version}</version>
+    </dependency>
+    -->
```
Comment (vuldin) [inline quarkus-reactive-kafka-consumer/pom.xml:75]: One example project I found required `apicurio-registry-utils-serde for serialization/deserialization. The `apicurio.version` was incompatible with this dependency so I commented this code out. It may be needed once we start reading the events in the consumer code.

File: quarkus-reactive-kafka-consumer/src/main/resources/application.properties [hunk 4]
```diff
@@ -6,17 +6,18 @@ quarkus.log.console.color=true
...
+%prod.mp.messaging.connector.smallrye-kafka.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username=${KAFKA_SCRAM_USERNAME} password=${KAFKA_SCRAM_PASSWORD};
+%prod.mp.messaging.connector.smallrye-kafka.ssl.truststore.location=${KAFKA_SSL_TRUSTSTORE_LOCATION}
+%prod.mp.messaging.connector.smallrye-kafka.ssl.truststore.password=${KAFKA_SSL_TRUSTSTORE_PASSWORD}
+%prod.mp.messaging.connector.smallrye-kafka.ssl.truststore.type=PKCS12
 # configuration to read messages
-mp.messaging.connector.smallrye-kafka.bootstrap.servers=${KAFKA_BOOTSTRAP_SERVERS}
-mp.messaging.incoming.my-data-stream.connector=smallrye-kafka
-mp.messaging.incoming.my-data-stream.topic=${KAFKA_TOPIC}
-mp.messaging.incoming.my-data-stream.value.deserializer=org.apache.kafka.common.serialization.DoubleDeserializer
+#mp.messaging.connector.smallrye-kafka.bootstrap.servers=${KAFKA_BOOTSTRAP_SERVERS}
```
Comment (vuldin) [inline quarkus-reactive-kafka-consumer/src/main/resources/application.properties:18]: Removed this line since it gets defaulted to the proper value if it doesn't exist.

**Comment by efir-tractatus** [comment]
Hey Josh,

The latest changes to the kafka consumer application.properties were merged into the main branch. The consumer itself though is very barebones and your implementation of the consumer code should be a better starting point.
Can you update your branch with the latest from main and reimplement yor consumer code?

**Comment by vuldin** [comment]
> Hey Josh,
> 
> The latest changes to the kafka consumer application.properties were merged into the main branch. The consumer itself though is very barebones and your implementation of the consumer code should be a better starting point. Can you update your branch with the latest from main and reimplement yor consumer code?

Thanks for the heads up! I really need to come back to this, and I will definitely make sure to pull the latest from main branch once I do.
